### DISCUSSION TITLE
Why does initialization always need to "app = FastAPI()"

### DISCUSSION THREAD

**Discussion by touero** [fastapi/fastapi#10466] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

@FastAPI.get("/")
async def test():
    return {result: "test"}
```

### Description

Sometimes we have to use apps to decorate our functions.

May I ask if the above writing method is supported and if there are any issues.

app.get('/') will access the wrong page.

I am trying to inherit the FastAPI,  I tried to write it as a static method, but self cannot access it anymore.

### Operating System

Linux, Windows

### FastAPI Version

0.95.1

### Pydantic Version

1.10.12

### Python Version

Python 3.9.13

**Comment by Kludex** [comment, marked as answer]
It's not supported.

**Reply by touero** [reply]
Can we achieve this goal by subclass from the FastAPI ?

**Comment by touero** [comment]
from fastapi import FastAPI

class MyApp(FastAPI):
    def __init__(self):
        super().__init__()
    
    @staticmethod
    def get(*args, **kwargs):
        super().get(*args, **kwargs)

@MyApp.get("/")
async def test():
    return {result: "test"}
